RFP-01.md

```diff
@@ -0,0 +1,183 @@
+    // --- view functions ---
+
+    /**
+     * View function for determining the minimum collateral required if using `token`
+     * If the token is not a valid collateral, this function reverts.
+     * @param token The token to retrieve the collateral requirement for
+     */
+    function collateral(IERC20 token) external view returns (uint256);
+    /**
+     * Determine if an address has a collateralized sub-pool and has not announced an exit.
+     * While not enforced on-chain, an address for which this call returns `true` is a
+     * solver and is allowed to submit settlement on CoW Protocol through
+     * `SignedSettlement`.
+     * @param candidate The address to check
+     * @return bool if the address should be allowed to submit settlements through
+     * `SignedSettlement`
+     */
+    function canSolve(address candidate) external view returns (bool);
+    /**
+     * View function for determining the sub-pool for a given solver.
+     * Reverts if the input address has no sub-pool to its name.
+     * @param solver The solver to retrieve the sub-pool for
+     * @return The contract managing the collateral paid by this solver to be
+     * allowed to settle transactions on CoW Protocol
+     */
+    function solverSubPool(address solver) external view returns (SubPool);
+    /**
+     * View function for determining the backend URI for the sub-pool
+     * @return string URI
+     */
+    function backendUri(SubPool pool) external view returns (string memory);
+
+    // --- permissionless functions ---
+    /**
+     * Deploy a `SubPool` deterministically based on `msg.sender`.
+     * @notice The `msg.sender` SHOULD NOT be a contract.
+     * @param token The nominated token to use for collateral
+     * @param amt How much collateral to add to the sub-pool (the call will revert if
+     * this value is below the minimal collateral needed for `token`)
+     * @param cowAmt Amount of COW to add as collateral
+     * @param backendUri The URI to send the batches to the solver
+     * @dev Emits an event `SubPoolCreated` when creating a sub-pool
+     */
+    function create(IERC20 token, uint256 amt, uint256 cowAmt, string calldata backendUri) external;
+
+    // --- cow dao bonding pool authed functions ---
+
+    /**
+     * Set the minimum delay before a solver can claim back the collateral from its
+     * sub-pool
+     * @dev Emits an event when setting a delay
+     * @param delay The delay in seconds
+     */
+    function setExitDelay(uint256 delay) external auth;
+    /**
+     * Allow `token` to be used as collateral for the sub-pool.
+     * @dev Emits an event `AllowCollateral` when allowing/updating a token
+     */
+    function allowCollateral(IERC20 token, uint256 min) external auth;
+    /**
+     * Revoke `token` as a collateral for sub-pools
+     * @dev Emits an event `RevokeCollateral` when revoking a token
+     */
+    function revokeCollateral(IERC20 token) external auth;
```

Comment:
This means that there should also be functions in the pool that allow pulling out "old" collateral and replacing it with new collateral.

Reply:
I'm not so sure, I think this is muddied a bit by the 365 day gradual accumulation towards the target deposit bond, in which case then it's not clear as to how to determine the minimum collateral. I'd think in this (extreme) case, it would be advisable that they just exit and create a new subpool.

Reply:
We might wanna consider the exit delay for invalidated collateral then and if it would allow a more faster exit.

Reply:
This is no longer the case as what is, and is not, acceptable collateral will be handled off-chain. Collateralisation (`canSolve`) is only affected by if there is an outstanding bill or not.